### PR TITLE
Drop gcloud SDK from buildbox

### PR DESCRIPTION
**Pull request by codingllama**
It's not used by any automated system.

If/when we do reinstate tests that use it, we should prefer a separate container.

### CODE REVIEW

File: build.assets/Dockerfile [hunk 1]
```diff
@@ -111,7 +111,6 @@ ENV LANGUAGE="en_US.UTF-8" \
 ARG BUILDARCH
 
 # Install packages.
-# Java JRE is required by gcloud firestore emulator.
```
Comment (codingllama) [inline build.assets/Dockerfile:114]: Is this comment the true? We'll see. :D

**Review by jakule — COMMENTED** [review]
I'm not sure that this is not being used https://github.com/gravitational/teleport/blob/0ae14dcb4d060bb103a9fa60adaac40263fa2e84/build.assets/Makefile#L279

File: build.assets/Dockerfile (hunk 1 above)
Comment (jakule) [reply build.assets/Dockerfile:114]: I know that when you remove it, the Firestore emulator stops working. AFAIK nothing else requires Java, I hope so.

**Comment by codingllama** [comment]
@jakule thanks, let me put that to the test.

**Comment by jakule** [comment]
> > I'm not sure that this is not being used
> > https://github.com/gravitational/teleport/blob/0ae14dcb4d060bb103a9fa60adaac40263fa2e84/build.assets/Makefile#L279
> 
> I don't see anything ever setting `TELEPORT_FIRESTORE_TEST`

I don't think we check the environment variable anymore
https://github.com/gravitational/teleport/blob/ba0897311c41bc9a72964664aedefc82061fa020/lib/events/firestoreevents/firestoreevents_test.go#L129C1-L144
If the emulator is not running, all these tests should be skipped. I don't see any firestore test skipped in out CI.

I agree with @codingllama We should check it.

File: build.assets/Dockerfile
Comment (codingllama) [inline build.assets/Dockerfile:1]: @jakule @rosstimothy moving convo here so we get a proper thread.

Looking at the CI jobs that ran for this very PR (remember that it runs the _old_ buildbox):

```
$ rg -i 'firestore tests'
logs_1051223/1_Unit Tests (Go) (1).txt
1828:2023-08-17T20:01:21.0321556Z     firestorebk_test.go:100: Firestore tests are disabled. Enable by defining the TELEPORT_FIRESTORE_TEST environment variable
1831:2023-08-17T20:01:21.0322503Z     firestorebk_test.go:100: Firestore tests are disabled. Enable by defining the TELEPORT_FIRESTORE_TEST environment variable

logs_1051223/Unit Tests (Go)/6_Run tests.txt
1004:2023-08-17T20:01:21.0321551Z     firestorebk_test.go:100: Firestore tests are disabled. Enable by defining the TELEPORT_FIRESTORE_TEST environment variable
1007:2023-08-17T20:01:21.0322499Z     firestorebk_test.go:100: Firestore tests are disabled. Enable by defining the TELEPORT_FIRESTORE_TEST environment variable
```

So it looks like ./lib/backend/firestore tests are already not running.

I'm trying ./lib/events/firestoreevents, and a few others, in a "new" buildbox. I'll report back once it completes.

Comment (codingllama) [reply build.assets/Dockerfile:1]: Scratch that, lib/events ain't running either:

```
$ rg -i 'firestore emulator'
logs_1051223/1_Unit Tests (Go) (1).txt
1905:2023-08-17T20:01:21.0342646Z     firestoreevents_test.go:46: Firestore emulator is not running, start it with: gcloud beta emulators firestore start --host-port=localhost:8618

logs_1051223/Unit Tests (Go)/6_Run tests.txt
1081:2023-08-17T20:01:21.0342643Z     firestoreevents_test.go:46: Firestore emulator is not running, start it with: gcloud beta emulators firestore start --host-port=localhost:8618
```

Comment (codingllama) [reply build.assets/Dockerfile:1]: Friendly ping @jakule @rosstimothy ?

**Review by rosstimothy — APPROVED** [review]

File: build.assets/Dockerfile [hunk 2]
```diff
@@ -184,12 +182,6 @@ RUN --mount=type=bind,target=/context \
...
     && rm osslsigncode.zip
 
-# Install gcloud SDK and Firestore emulator.
-ENV PATH="$PATH:/opt/google-cloud-sdk/bin"
-RUN (curl -fsSL https://sdk.cloud.google.com | bash -s -- --install-dir=/opt --disable-prompts) && \
-    gcloud components install cloud-firestore-emulator beta && \
-    rm -rf /opt/google-cloud-sdk/.install/.backup
-
 # Install etcd.
 RUN curl -fsSL https://github.com/coreos/etcd/releases/download/v3.3.9/etcd-v3.3.9-linux-${BUILDARCH}.tar.gz | tar -xz && \
```
[the remark this answers is not in the document]
Comment (codingllama) [reply build.assets/Dockerfile:186]: Sounds like a follow up to me.

**Comment by codingllama** [comment]
Friendly ping @jakule ?

**Review by jakule — APPROVED** [review]

File: build.assets/Dockerfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (codingllama) [reply build.assets/Dockerfile:186]: https://github.com/gravitational/teleport/pull/30640

**Pull request merged**